dot/core/service.go

```diff
@@ -544,56 +544,58 @@ func (s *Service) createBlockResponse(msg *network.BlockRequestMessage) (*networ
 	responseData := []*types.BlockData{}
 
 	for _, hash := range subchain {
-		data, err := s.blockState.GetBlockData(hash)
-		if err != nil {
-			return nil, err
-		}
 
 		blockData := new(types.BlockData)
 		blockData.Hash = hash
 
-		// TODO: checks for the existence of the following fields should be implemented once #596 is addressed.
+		// set defaults
+		blockData.Header = optional.NewHeader(false, nil)
+		blockData.Body = optional.NewBody(false, nil)
+		blockData.Receipt = optional.NewBytes(false, nil)
+		blockData.MessageQueue = optional.NewBytes(false, nil)
+		blockData.Justification = optional.NewBytes(false, nil)
 
 		// header
-		if msg.RequestedData&1 == 1 {
-			blockData.Header = data.Header
-		} else {
-			blockData.Header = optional.NewHeader(false, nil)
+		if (blockRequest.RequestedData & 1) == 1 {
```

Comment:
remove spaces for consistency, same with below

Reply:
well, goimports things otherwise: `dot/core/service.go:559: File is not `goimports`-ed (goimports)` so I have to leave it like this.

Reply:
okay that's weird :/ not sure why it's ok with the other ones then